src/encoded/types/patient.py

```diff
@@ -466,51 +475,62 @@ def surgery_summary(self, request, surgery=None):
+                    germline,ihc, consent,medical_imaging,supportive_medication, diagnosis_date_tumor_registry=None):
         nephrectomy_dates = []
         non_nephrectomy_dates = []
+        surgery_dates = []
         diagnosis_date = "Not available"
+        diagnosis_source = "Not applicable"
+
+        # Calculate
         if len(surgery) > 0:
             for surgery_record in surgery:
                 surgery_object = request.embed(surgery_record, '@@object')
-                surgery_procedure = surgery_object['surgery_procedure']
-                for procedure_record in surgery_procedure:
-                    procedure_obj = request.embed(procedure_record, '@@object')
-                    if procedure_obj['procedure_type'] == "Nephrectomy":
+                surgery_path_report = surgery_object['pathology_report']
+                for path_report in surgery_path_report:
+                    path_report_obj = request.embed(path_report, '@@object')
+                    if path_report_obj['path_source_procedure'] == "path_nephrectomy":
                         nephrectomy_dates.append(surgery_object['date'])
-                    elif  procedure_obj['procedure_type'] == "Biopsy" or procedure_obj['procedure_type'] == "Metastectomy":
+                    elif  path_report_obj['path_source_procedure'] == "path_biopsy" or path_report_obj['path_source_procedure'] == "path_metasis":
                         non_nephrectomy_dates.append(surgery_object['date'])
+
         if len(nephrectomy_dates) > 0 :
-            nephrectomy_dates.sort(key = lambda date: datetime.strptime(date, '%Y-%m-%d'))
-            diagnosis_date = nephrectomy_dates[0]
-        else:
-            if len(radiation) > 0:
-                # add radiation dates
-                for radiation_record in radiation:
-                    radiation_object = request.embed(radiation_record, '@@object')
-                    non_nephrectomy_dates.append(radiation_object['start_date'])
-            if len(medication) > 0:
-                # add medication dates
-                for medication_record in medication:
-                    medication_object = request.embed(medication_record, '@@object')
-                    non_nephrectomy_dates.append(medication_object['start_date'])
-
-            if len(non_nephrectomy_dates) > 0:
+                nephrectomy_dates.sort(key = lambda date: datetime.strptime(date, '%Y-%m-%d'))
+                diagnosis_date = nephrectomy_dates[0]
+                surgery_dates.append(nephrectomy_dates[0])
+                diagnosis_source = "Pathology Report"
+
+        elif len(non_nephrectomy_dates) > 0:
+            non_nephrectomy_dates.sort(key = lambda date: datetime.strptime(date, '%Y-%m-%d'))
+            surgery_dates.append(non_nephrectomy_dates[0])
+            surgery_dates.sort(key = lambda date: datetime.strptime(date, '%Y-%m-%d'))
+            diagnosis_date = surgery_dates[0]
```

Comment:
surgery_dates is not needed, diagnosis_date get be calculated by just using nephrectomy_dates and non_nephrectomy_dates.

Reply:
It just to find the earliest date between nephrectomy and biopsy. So made an intermediate surgery date to sort those.

Adding comment to clarify

Reply:
should use if if to check nephrectomy_dates and non_nephrectomy_dates, instead of if elif